tests/php/tests/includes/test_class.wp-job-manager-data-exporter.php

```diff
@@ -0,0 +1,194 @@
+<?php
+/**
+ * Defines a class to test the data exporter
+ *
+ * @package wp-job-manager-test
+ * @since 1.31.1
+ */
+
+/**
+ * Handles the user data export.
+ *
+ * @package
+ * @since
+ */
+class WP_Job_Manager_Data_Exporter_Test extends WPJM_BaseTest {
+	/**
+	 * Setup user metadata
+	 *
+	 * @param array $args The metadata to be added.
+	 * @param array $expected The expected output.
+	 */
+	private function setupUserMeta( $args, &$expected ) {
+		$user_id = $this->factory()->user->create(
+			array(
+				'user_login' => 'johndoe',
+				'user_email' => 'johndoe@example.com',
+				'role'       => 'subscriber',
+			)
+		);
+
+		if ( isset( $args['_company_logo'] ) ) {
+			$args['_company_logo'] = $this->factory()->post->create(
+				array( 'post_type' => 'attachment' )
+			);
+			$expected['data']['data']['Company Logo']['value'] = $args['_company_logo'];
+		}
+
+		foreach ( $args as $key => $value ) {
+			update_user_meta( $user_id, $key, $value );
+		}
+	}
+
+	/**
+	 * Test the data exporter for an invalid email
+	 */
+	public function test_data_exporter_for_invalid_email() {
+		// ARRANGE.
+		$exporter = new WP_Job_Manager_Data_Exporter();
+
+		// ASSERT.
+		$result = $exporter->user_data_exporter( 'this-is-an-invalid-email' );
+
+		// ACT.
+		$this->assertEmpty( $result );
+	}
+
+	/**
+	 * Test the user data exporter method
+	 *
+	 * @dataProvider data_provider
+	 * @param array $args The metadata to be added.
+	 * @param array $expected The expected output.
+	 */
+	public function test_user_data_exporter( $args, $expected ) {
+		// ARRANGE.
+		$this->setupUserMeta( $args, $expected );
+		$exporter = new WP_Job_Manager_Data_Exporter();
+		if ( $id = email_exists( 'johndoe@example.com' ) ) {
```

Comment:
Could this be broken out into two separate lines for better readability?

Something like:

```
$id = email_exists( 'johndoe@example.com' );
if ( false !== $id ) {
```